edge-util/src/Microsoft.Azure.Devices.Edge.Storage.RocksDb/disk/DiskSpaceChecker.cs

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Storage.RocksDb.Disk
+{
+    using System;
+    using System.IO;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Storage.Disk;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public class DiskSpaceChecker : IStorageSpaceChecker
+    {
+        static readonly int checkFrequencyDefault = 120;
+        readonly string storageFolder;
+        readonly object updateLock = new object();
+        Option<DiskSpaceCheckerBase> inner;
+        long maxSize;
+        Option<int> checkFrequency;
+
+        DiskSpaceChecker(string storageFolder, Option<DiskSpaceCheckerBase> inner)
+        {
+            this.storageFolder = Preconditions.CheckNonWhiteSpace(storageFolder, nameof(storageFolder));
+            this.inner = Preconditions.CheckNotNull(inner, nameof(inner));
+        }
+
+        // Returns false if checker is disabled (i.e. Option.HasValue is false)
+        public bool IsFull => this.inner.Match<bool>(b => b.DiskStatus == DiskSpaceStatus.Full, () => false);
+
+        public static DiskSpaceChecker Create(string storageFolder)
+        {
+            // Start up diskSpaceChecker unfilled - so that it will be enabled when SetMaxSizeBytes is called
+            var diskSpaceChecker = new DiskSpaceChecker(storageFolder, Option.None<DiskSpaceCheckerBase>());
+            return diskSpaceChecker;
+        }
+
+        public void SetMaxSizeBytes(long maxSizeBytes)
+        {
+            this.maxSize = maxSizeBytes;
+            this.inner.ForEach(b => b.SetMaxSizeBytes(maxSizeBytes));
```

Comment:
Instead of doing this I would change it to the following as you shouldn't expect this.inner to always be a FixedSizeDiskSpaceChecker given how someone might be switching from a different checker to the fixed size one:

1. Dispose original checker -> this.inner?.Dispose(). This will take care of stopping the old checker.
2. Set this.inner to a new FixedSizeDiskSpaceChecker instance.